src/LogicBlock/SMTLibLogicModel.cpp

```diff
@@ -0,0 +1,131 @@
+
+#include "LogicBlock/SMTLibLogicModel.hpp"
+
+#include <regex>
+
+std::string smtliblogic::SMTLibLogicModel::getTermString(const LogicTerm& a) {
+    std::stringstream ss(modelString);
+    std::string       line;
+
+    const auto  regex = std::regex("\\(define-fun (.*) \\(\\) (.*) (.*)\\)");
+    std::smatch m;
+    while (std::getline(ss, line)) {
+        if (std::regex_search(line, m, regex)) {
+            if (m.size() == 4) {
+                if (m[1] == a.getName()) {
+                    return line;
+                }
+            }
+        }
+    }
+    return {};
+}
+std::string smtliblogic::SMTLibLogicModel::getTermValue(const logicbase::LogicTerm& a) {
+    std::string const termString = getTermString(a);
+
+    const auto  regex = std::regex(R"(\(define-fun (.*) \(\) (.*) (.*)\))");
+    std::smatch m;
+
+    if (std::regex_search(termString, m, regex)) {
+        if (m.size() == 4) {
+            return m[3];
+        }
+    }
+
+    return {};
+}
+int smtliblogic::SMTLibLogicModel::getIntValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
+    std::string const value = getTermValue(a);
+    const auto        type  = getTermType(a);
+    if (type != "Int") {
+        throw std::runtime_error("SMTLibLogicModel::getIntValue: type mismatch");
+    }
+    if (value.empty()) {
+        return 0;
+    }
+    return std::stoi(value);
+}
+bool smtliblogic::SMTLibLogicModel::getBoolValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
+    std::string const value = getTermValue(a);
+    const auto        type  = getTermType(a);
+    if (type != "Bool") {
+        throw std::runtime_error("SMTLibLogicModel::getBoolValue: type mismatch");
+    }
+    if (value.empty()) {
+        return false;
+    }
+    return value == "true";
+}
+double smtliblogic::SMTLibLogicModel::getRealValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
+    std::string const value = getTermValue(a);
+    const auto        type  = getTermType(a);
+    if (type != "Real") {
+        throw std::runtime_error("SMTLibLogicModel::getRealValue: type mismatch");
+    }
+    if (value.empty()) {
+        return 0.0;
+    }
+    return std::stod(value);
+}
+uint64_t smtliblogic::SMTLibLogicModel::getBitvectorValue(const logicbase::LogicTerm& a, logicbase::LogicBlock* lb) {
+    std::string const value = getTermValue(a);
+    if (value.empty()) {
+        return 0;
+    }
+    const auto type = getTermType(a);
+
+    const auto  typeRegex = std::regex(R"(\(_ BitVector (\d+)\))");
```

Comment:
As noted above, regex construction in C++ is pretty expensive. Consider making this either a protected class member, or a static const variable here so that it is only constructed once.